### PR TITLE
Added screen rotation support to JavaCamera2View amd NativeCameraView. Fixed JavaCamera2View initialization.

### PR DESCRIPTION
**Pull request by alexlyulkov**
Added automatic image rotation to JavaCamera2View and NativeCameraView so the video preview was matched with screen orientation.
Fixed double preview initialization bug in JavaCamera2View.
Added proper cameraID parsing to NativeCameraView similar to JavaCameraView

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/java/generator/android-21/java/org/opencv/android/JavaCamera2View.java [hunk 1]
```diff
@@ -322,12 +332,16 @@ protected boolean connectCamera(int width, int height) {
                     mCaptureSession.close();
                     mCaptureSession = null;
                 }
-                createCameraPreviewSession();
```
Comment (alexlyulkov) [inline modules/java/generator/android-21/java/org/opencv/android/JavaCamera2View.java:325]: There is already a createCameraPreviewSession call in onOpened camera callback.

File: modules/java/generator/android-21/java/org/opencv/android/JavaCamera2View.java [hunk 2]
```diff
@@ -213,8 +211,9 @@ public void onImageAvailable(ImageReader reader) {
                     assert (planes.length == 3);
                     assert (image.getFormat() == mPreviewFormat);
 
-                    JavaCamera2Frame tempFrame = new JavaCamera2Frame(image);
+                    RotatedCameraFrame tempFrame = new RotatedCameraFrame(new JavaCamera2Frame(image), mFrameRotation);
                     deliverAndDrawFrame(tempFrame);
+                    tempFrame.mFrame.release();
```
Comment (asmorkalov) [inline modules/java/generator/android-21/java/org/opencv/android/JavaCamera2View.java:216]: I think it should be moved inside RotatedCameraFrame.release().

**Comment by asmorkalov** [comment]
@alexlyulkov I do not see preview with Pixel8 in Tutorial 4. Just black screen with text status in corner.

File: modules/java/generator/android-21/java/org/opencv/android/JavaCamera2View.java (hunk 1 above)
Comment (asmorkalov) [reply modules/java/generator/android-21/java/org/opencv/android/JavaCamera2View.java:325]: Internal native cv::Mat works as smart pointer. Even, if the creator removed Java object the natve object is not removed and hold by this reference.

**Comment by asmorkalov** [comment]
Tutorial1-Camera-Preview works with all back-ends JavaCameraView, Java2CameraView and NativeCameraView with rotation. Please ignore the previous comment on Tutorial4. The sample does not work without the patch. It should be fixed separately.

File: modules/java/generator/android-24/java/org/opencv/android/NativeCameraView.java [hunk 3]
```diff
@@ -90,27 +91,59 @@ public int getHeight(Object obj) {
...
             if (mCamera == null)
                 return false;
 
             if (mCamera.isOpened() == false)
                 return false;
 
-            mFrame = new NativeCameraFrame(mCamera);
+            android.hardware.Camera.CameraInfo info = new android.hardware.Camera.CameraInfo();
+            android.hardware.Camera.getCameraInfo(localCameraIndex, info);
+            int frameRotation = getFrameRotation(
```
Comment (asmorkalov) [inline modules/java/generator/android-24/java/org/opencv/android/NativeCameraView.java:131]: Looks like you can re-use CameraInfo from cameras iteration when camera is selected.

File: modules/java/generator/android/java/org/opencv/android/JavaCameraView.java [hunk 4]
```diff
@@ -245,7 +249,9 @@ protected void releaseCamera() {
                 mFrameChain[1].release();
             }
             if (mCameraFrame != null) {
+                mCameraFrame[0].mFrame.release();
```
Comment (asmorkalov) [inline modules/java/generator/android/java/org/opencv/android/JavaCameraView.java:252]: The same idea on release()

File: modules/java/generator/android-24/java/org/opencv/android/NativeCameraView.java [hunk 5]
```diff
@@ -131,7 +164,10 @@ private boolean initializeCamera(int width, int height) {
 
     private void releaseCamera() {
         synchronized (this) {
-            if (mFrame != null) mFrame.release();
+            if (mFrame != null) {
+                mFrame.mFrame.release();
```
Comment (asmorkalov) [inline modules/java/generator/android-24/java/org/opencv/android/NativeCameraView.java:168]: The same for release()

File: modules/java/generator/android-21/java/org/opencv/android/JavaCamera2View.java (hunk 2 above)
Comment (alexlyulkov) [reply modules/java/generator/android-21/java/org/opencv/android/JavaCamera2View.java:216]: I thought about this too, but I think it is not good to release an object inside a class if the object is created outside. RotatedCameraFrame constructor takes a pointer to existing JavaCamera2Frame object, and this object may be used in other place.

**Pull request merged**